hive/src/main/java/io/confluent/connect/storage/hive/HiveSchemaConverter.java

```diff
@@ -90,6 +98,29 @@ public static TypeInfo convertMap(Schema schema) {
   }
 
   public static TypeInfo convertPrimitive(Schema schema) {
-    return TYPE_TO_TYPEINFO.get(schema.type());
+    if (schema.name() == null) {
+      return TYPE_TO_TYPEINFO.get(schema.type());
+    }
+
+    //TODO: should take Hive version into consideration
```

Comment:
See comment about defining new config properties above, that will allow you to enable or disable this feature